src/mapping/LinearCellInterpolationMapping.cpp

```diff
@@ -0,0 +1,100 @@
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  } else {
+    PRECICE_ASSERT(constraint == SCALEDCONSISTENT, constraint);
+    setInputRequirement(Mapping::MeshRequirement::FULL);
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  }
+
+  PRECICE_CHECK(constraint != SCALEDCONSISTENT, "Volume mapping doesn't support scaled-consistent mappings.");
+}
+
+void LinearCellInterpolationMapping::computeMapping()
+{
+  PRECICE_TRACE(input()->vertices().size(), output()->vertices().size());
+  const std::string     baseEvent = "map.vci.computeMapping.From" + input()->getName() + "To" + output()->getName();
+  precice::utils::Event e(baseEvent, precice::syncMode);
+  PRECICE_ASSERT(getDimensions() == 2, "Volume mapping not available in 3D.");
+
+  // Setup Direction of Mapping
+  mesh::PtrMesh origins, searchSpace;
+  if (hasConstraint(CONSERVATIVE)) {
+    PRECICE_DEBUG("Compute conservative mapping");
+    origins     = input();
+    searchSpace = output();
+  } else {
+    PRECICE_DEBUG("Compute consistent mapping");
+    origins     = output();
+    searchSpace = input();
+  }
+
+  const auto &fVertices = origins->vertices();
+
+  if (getDimensions() == 2) {
+    if (!fVertices.empty() && searchSpace->triangles().empty()) {
+      PRECICE_WARN("2D Mesh \"{}\" does not contain triangles. "
+                   "Linear cell interpolation falls back to nearest projection mapping.",
+                   searchSpace->getName());
+    }
+  } else {
+    // TODO, shouldn't be reached
+    PRECICE_ERROR("Linear Cell interpolation in 3D not implemented yet. This line should be unreachable");
+  }
+
+  // Amount of nearest elements to fetch for detailed comparison.
+  // This safety margin results in a candidate set which forms the base for the
+  // local nearest projection and counters the loss of detail due to bounding box generation.
+  // @TODO Add a configuration option for this factor
+  constexpr int nnearest = 4;
+
+  auto &                                 index = searchSpace->index();
+  utils::statistics::DistanceAccumulator fallbackStatistics;
+
+  _interpolations.clear();
+  _interpolations.reserve(fVertices.size());
+
+  for (const auto &fVertex : fVertices) {
+    // Find tetrahedra (3D) or triangle (2D) or fall-back on NP
+    auto match = index.findCellOrProjection(fVertex.getCoords(), nnearest);
+    _interpolations.push_back(std::move(match.polation));
+    if (!math::equals(match.distance, 0.0)) {
+      // Only push when fall-back occurs, so the number of entries is the number of vertices outside the domain
+      fallbackStatistics(match.distance);
+    }
+  }
+
+  if (!fallbackStatistics.empty()) {
+    PRECICE_INFO("Some points are outisde of the domain defined by connectivity. Fall-back on Nearest-Projection occured."
+                 " Projection distances: {}",
+                 fallbackStatistics);
+  } else {
+    PRECICE_INFO("All vertices are inside cells, no fallback to Nearest-Projection required");
+  }
```

Comment:
We need to improve the wording here still a bit. Not sure if a user would understand what is happening already.

Reply:
# Comment PR

Actually, an interesting thing occurs when we have no triangles but enough edges to always find a projection:

```
---[precice]  Compute "read" mapping from mesh "Flow-Mesh" to mesh "Chemistry-Mesh".
---[precice] WARNING:  2D Mesh "Flow-Mesh" does not contain triangles. Linear cell interpolation falls back to nearest projection mapping.
---[precice]  All vertices are inside cells, no fallback to Nearest-Projection required
```

Wording should be more consistent as having just edges trigger contradictory messages ("No triangles ! => then right afterwards "All projections were fine").

That's probably an edge case (no pun intended), as I got this with matching meshes and only edges configured, so projection was trivial. Maybe we should simply don't log anything if no fallback occurs? (But keep the `triangles.empty()` warning)

But back to a more standard case, with actual triangles but some projection, typical log is 

```
---[precice]  Compute "read" mapping from mesh "Mesh1" to mesh "Mesh2".
---[precice]  Some points are outside of the domain defined by connectivity. Fall-back on Nearest-Projection occured. Projection distances: min:0.1 max:0.1 avg: 0.1 var: 0 cnt: 2
```

So, in context, it is clear that it is part of the mapping computation, but not that this is Linear Cell Interpolation specific behavior.
Alternative text suggested could be something like

```
---[precice]  Compute "read" mapping from mesh "Mesh1" to mesh "Mesh2".
---[precice]  Linear Cell Interpolation is used, but some points from "Mesh2" don't lie in the domain defined by the triangles/tetrahedra. These points have been projected on the domain boundary. This could come from rounding errors or wrong connectivity configuration. If distances seem too large, please check your mesh. The projection statistics are: min:0.1 max:0.1 avg: 0.1 var: 0 cnt: 2
```

(where triangles/tetrahedra could be dynamically picked at runtime to show the corresponding on, and "Mesh2" replaced by target mesh name)

Also, should we keep it as PRECICE_INFO or make it a warning ?